perf_tests/large_scale_performance_locustfile.py

```diff
@@ -0,0 +1,250 @@
+            ia_code = search_category if search_category in DATA['available_product_codes'] else random.choice(list(DATA['ia_codes']))
+            score_modifiers = {
+                "add_to_score": [
+                    {"field_name": f"artist_sales_scores.{ia_code}", "weight": random.uniform(0.0001, 0.005)},
+                    {"field_name": f"recent_sales_scores.{ia_code}+{region}", "weight": random.uniform(0.0001, 0.005)},
+                    {"field_name": f"recent_sales_scores.{ia_code}+ALL_WORLD", "weight": random.uniform(0.0001, 0.005)},
+                    {"field_name": f"all_time_sales_scores.{ia_code}+ALL_WORLD", "weight": random.uniform(0.0001, 0.005)},
+                    {"field_name": f"ro_scores.{search_keywords}+{ia_code}", "weight": random.uniform(0.0001, 0.005)},
+                ]
+            }
+            # Add filter on available_product_codes
+            filter_string = f"available_product_codes:{ia_code}"
+
+        # Construct the search parameters
+        search_params = {
+            'q': q,
+            'limit': 20,
+            'search_method': 'HYBRID', 
+            'hybrid_parameters': {
+                "retrievalMethod": "disjunction",
+                "rankingMethod": "rrf",
+                "alpha": 0.3,
+                "rrfK": 60,
+                "searchableAttributesLexical": ["tags"],
+                "scoreModifiersTensor": score_modifiers,
+                "scoreModifiersLexical": {
+                    "add_to_score": [
+                        {"field_name": f"artist_sales_scores.{ia_code}", "weight": random.uniform(0.1, 5)},
+                        {"field_name": f"recent_sales_scores.{ia_code}+{region}", "weight": random.uniform(0.1, 5)},
+                        {"field_name": f"recent_sales_scores.{ia_code}+ALL_WORLD", "weight": random.uniform(0.1, 5)},
+                        {"field_name": f"all_time_sales_scores.{ia_code}+ALL_WORLD", "weight": random.uniform(0.1, 5)},
+                        {"field_name": f"ro_scores.{search_keywords}+{ia_code}", "weight": random.uniform(1, 300)},
+                    ]
+                },
+            },
+            'attributes_to_retrieve': ['work_id', 'tags', 'available_product_codes'],
+        }
+        if filter_string:
+            search_params['filter_string'] = filter_string
+
+        # Perform the search and capture telemetry data
+        start_time = time.time()
+        try:
+            response = self.client.index(INDEX_NAME).search(**search_params)
+            # Print the response for debugging
+            #print("DEBUG: Search Response")
+            #print(json.dumps(response, indent=2))
+            total_time = (time.time() - start_time) * 1000  # in ms
+            # Extract telemetry data
+            telemetry = response.get('telemetry', {})
+            telemetry['total_time_ms'] = total_time
+            self.telemetry_data.append(telemetry)
```

Comment:
How large is this data, would it cause out of memory issue if we run this tests for a long time with large load? Should we consider append it to a file regularly to reduce the memory footprint?